.github/workflows/ci-pull-requests.yml

```diff
@@ -13,18 +13,27 @@ jobs:
         node-version: [14.x, 16.x]
 
     steps:
-      - uses: actions/checkout@v1
+      - uses: actions/checkout@v3
+
       - name: Use Node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v3
         with:
           node-version: ${{ matrix.node-version }}
           cache: 'npm'
-      - name: npm install, build, and test
-        run: |
-          npm install -g npm@latest
-          npm install
-          npm run build --if-present
-          npm test
-          npm run lint
+
+      - name: Install npm and dependencies
+        run: npm install -g npm@latest && npm install
+
+      - name: Build
+        run: npm run build --if-present
+
+      - name: Test
+        run: npm run test
+
+      - name: Fetch base branch for `git diff`
+        run: git fetch origin ${{ github.event.pull_request.base.ref }}:${{ github.event.pull_request.base.ref }}
+
+      - name: Run ESLint on changes only
         env:
-          CI: true
```

Comment:
I think `CI=true` is used by npm to ensure reproducible builds. It also disables npm telemetrry, and auditing (I think).

Reply:
I was going off of this: https://github.blog/changelog/2020-04-15-github-actions-sets-the-ci-environment-variable-to-true/

Set anyway?

Reply:
wasn't aware of that. makes sense.